src/cs/Ssh/Services/AuthenticationService.cs

```diff
@@ -475,8 +475,14 @@ protected override void Dispose(bool disposing)
 	{
 		if (disposing)
 		{
-			disposeCancellationSource.Cancel();
-			disposeCancellationSource.Dispose();
+			try
+			{
+				disposeCancellationSource.Cancel();
+				disposeCancellationSource.Dispose();
```

Comment:
There are no native resources used in `disposeCancellationSource` (e.g. wait handles)? If so, it's OK to not dispose it.

Reply:
It implements `IDisposable`, so the code analysis will warn if it is not disposed.